mods/cnc/maps/nod06b/nod06b.lua

```diff
@@ -0,0 +1,189 @@
+	if not Atk2Switch then
+		for type, count in pairs({ ['e2'] = 2, ['e1'] = 2}) do
+			MyActors = Utils.Take(count, GDI.GetActorsByType(type))
+			Utils.Do(MyActors, function(actor)
+				MoveAndHunt(actor, Gdi5Waypoint)
+			end)
+		end
+		Atk2Switch = true
+	end
+end
+
+Chn2TriggerFunction = function()
+	if not Chn2Switch then
+		local cargo = Reinforcements.ReinforceWithTransport(GDI, 'tran', Gdi1Units, Chn2Waypoints, { ChnEntry.Location })[2]
+		Utils.Do(cargo, function(actor)
+			IdleHunt(actor)
+		end)
+		Chn2Switch = true
+	end
+end
+
+MoveAndHunt = function(unit, waypoints)
+	if unit ~= nil then
+		Utils.Do(waypoints, function(waypoint)
+			unit.AttackMove(waypoint.Location)
+		end)
+		IdleHunt(unit)
+	end
+end
+
+InsertNodUnits = function()
+	Media.PlaySpeechNotification(Nod, "Reinforce")
+	Camera.Position = UnitsRallyVehicle2.CenterPosition
+
+	Reinforcements.Reinforce(Nod, NodUnitsVehicle1, { UnitsEntryVehicle.Location, UnitsRallyVehicle1.Location }, 10)
+	Reinforcements.Reinforce(Nod, NodUnitsVehicle2, { UnitsEntryVehicle.Location, UnitsRallyVehicle2.Location }, 15)
+	Reinforcements.Reinforce(Nod, NodUnitsGunner, { UnitsEntryGunner.Location, UnitsRallyGunner.Location }, 15)
+	Reinforcements.Reinforce(Nod, NodUnitsRocket, { UnitsEntryRocket.Location, UnitsRallyRocket.Location }, 25)
+end
+
+
+WorldLoaded = function()
+	GDI = Player.GetPlayer("GDI")
+	Nod = Player.GetPlayer("Nod")
+
+	Trigger.OnObjectiveAdded(Nod, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+
+	Trigger.OnObjectiveCompleted(Nod, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+
+	Trigger.OnObjectiveFailed(Nod, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+
+	Trigger.OnPlayerWon(Nod, function()
+		Media.PlaySpeechNotification(Nod, "Win")
+	end)
+
+	Trigger.OnPlayerLost(Nod, function()
+		Media.PlaySpeechNotification(Nod, "Lose")
+	end)
+
+	NodObjective1 = Nod.AddPrimaryObjective("Steal the GDI nuclear detonator")
+	GDIObjective = GDI.AddPrimaryObjective("Stop the Nod taskforce from escaping with the detonator")
+
+	InsertNodUnits()
+
+	Trigger.OnEnteredFootprint(HuntCellTriggerActivator, function(a, id)
+		if a.Owner == Nod then
+			HuntTriggerFunction()
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.OnEnteredFootprint(DzneCellTriggerActivator, function(a, id)
+		if a.Owner == Nod then
+			Actor.Create('flare', true, { Owner = Nod, Location = waypoint17.Location })
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.OnEnteredFootprint(Win1CellTriggerActivator, function(a, id)
+		if a.Owner == Nod then
+			Win1TriggerFunction()
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.OnEnteredFootprint(Win2CellTriggerActivator, function(a, id)
+		if a.Owner == Nod then
+			Nod.MarkCompletedObjective(NodObjective2)
```

Comment:
Theoretically, a player could fail to pick up the crate in the base and flee to the drop zone, thereby triggering this, err, trigger and causing a crash. You'll want `if a.Owner == Nod and NodObjective2 then`.

The same thing is true for `NodObjective3` in Nod06a. It would be great if you could fix that as well.